src/extensions/combat/combatext_hooks.cpp

```diff
@@ -94,6 +97,50 @@ DECLARE_PATCH(_Explosion_Damage_IsWallAbsoluteDestroyer_Patch)
 }
 
 
+/**
+ *  #issue-897
+ *
+ *  Implements IsIceDestruction scenario option for preventing destruction of ice,
+ *  and IceStrength for configuring the chance for ice getting destroyed.
+ *
+ *  @author: Rampastring
+ */
+DECLARE_PATCH(_Explosion_Damage_IsIceDestruction_Patch)
+{
+    GET_REGISTER_STATIC(const WarheadTypeClass *, warhead, edi);
+    GET_STACK_STATIC(int, strength, esp, 0x54);
+
+    if (!ScenExtension->IsIceDestruction) {
+        goto no_ice_destruction;
+    }
+
+    /**
+     *  Stolen bytes/code here.
+     */
+    if (warhead->IsWallDestroyer || warhead->IsConventional) {
+
+        /**
+         *  Allow destroying ice if the strength of ice is 0 or the random number check allows it.
+         */
+        if (RuleExtension->IceStrength <= 0 || Random_Pick(0, RuleExtension->IceStrength) < strength) {
```

Comment:
Can you check the output assembly for this in Release builds please? `Random_Pick` is an inline function and its possible the compiler might inline it using a different register than the expected `EAX` and `ECX` here.